File: service/src/main/java/com/spring/web/order/models/Service.java
package com.spring.web.order.models;

import javax.persistence.*;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name="services")
public class Service {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "category")
    private String category;

    @Column(name = "service_name")
    private String serviceName;

    @OneToMany(mappedBy = "service",
            fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Package> packages;

    @Column(name = "created_at", insertable = false, updatable = false)
    private Date createdAt;

    @Column(name = "updated_at", insertable = false, updatable = false)
    private Date updatedAt;

    public Service() {
        packages = new HashSet<>();
    }

    public Service(
            Long id,
            String category,
            String serviceName,
            Set<Package> packages
    ) {
        this.id = id;
        this.category = category;
        this.serviceName = serviceName;
        this.packages = packages;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getServiceName() {
        return serviceName;
    }

    public void setServiceName(String serviceName) {
        this.serviceName = serviceName;
    }

    public Set<Package> getPackages() {
        return packages;
    }

    public void setPackages(Set<Package> packages) {
        this.packages = packages;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: web/src/models/order/service.js
class ServiceData {
    constructor() {
        this.id = null;
        this.category = null;
        this.serviceName = null;
        this.items = null;
        this.createdAt = null;
        this.updatedAt = null;
    }

    getTableKeyList = () => {
        return ["id", "serviceName", "category"];
    }

    getLabelList = () => {
        return ["#", "Service name", "Category"];
    }

    getObjectData = () => {
        return {
            id: this.id,
            category: this.category,
            serviceName: this.serviceName,
            items: this.items,
            createdAt: this.createdAt,
            updatedAt: this.updatedAt
        }
    }

    setObjectData = (data) => {
        this.id = data.id;
        this.category = data.category;
        this.serviceName = data.serviceName;
        this.items = data.items;
        this.createdAt = data.createdAt;
        this.updatedAt = data.updatedAt;
    }
}

export default ServiceData;